### PR TITLE
THRUSTERS, ALARMS, LAUNCH

### PR DESCRIPTION
**Pull request by mattlangford**
The first PR to get the sub operable again
It includes:
- Porting Navigator Alarms back into the sub repo. This consists mainly of small changes here and there. Let me know if you see any `navigator`'s still in the diff.
- Allowing the thruster driver node to not crash if a thruster isn't physically plugged in. It will be treated just like a thruster that is dropped during operation and should be handled appropriately. In lab tests showed the system should be working.

### CODE REVIEW

File: command/sub8_alarm/nodes/event_listeners/keepalive_broadcaster.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        self.timer = rospy.Timer(rospy.Duration(0.1), self.keepalive_pub)
+
+    def keepalive_pub(self, *args):
+        h = sub8_ros_tools.make_header()
+        self.pub.publish(h)
+
+
+if __name__ == '__main__':
+    rospy.init_node('network_keepalive_broadcaster')
+    k = KeepAliveBroadcaster()
```
Comment (tessbianchi) [inline command/sub8_alarm/nodes/event_listeners/keepalive_broadcaster.py:25]: This replaces network_listener, I'm guessing?

File: command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py [hunk 2]
```diff
@@ -11,16 +11,23 @@ def __init__(self):
...
+            self.dropped_thrusters.append(thruster_name)
         # Drop a single thruster
         rospy.logwarn("Dropping thuster {}".format(thruster_name))
         # a priori: Each thruster name is a string, do not need to map(str, remaining_thursters)
         rospy.logwarn("Dropped thrusters: " + ', '.join(self.dropped_thrusters))
 
-    def handle(self, time_sent, parameters):
+    def handle(self, time_sent, parameters, alarm_name):
         self.drop_thruster(parameters['thruster_name'])
         self.update_layout(self.dropped_thrusters)
```
Comment (tessbianchi) [inline command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py:28]: Why add param that isn't being used? Will it be used in the future?

Comment (tessbianchi) [reply command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py:28]: Also, this doesn't match _template above.

File: command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py [hunk 3]
```diff
@@ -11,16 +11,23 @@ def __init__(self):
...
 
-    def cancel(self, time_sent, parameters):
-        self.dropped_thrusters.pop(self.dropped_thrusters.index(parameters['thruster_name']))
-        self.update_layout(self.dropped_thrusters)
+    def cancel(self, time_sent, parameters, alarm_name):
```
Comment (tessbianchi) [inline command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py]: Same comment as above

File: command/sub8_alarm/sub8_alarm/alarm_handlers/network_loss.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+import rospy
+from _template import HandlerBase
+from navigator_msgs.srv import WrenchSelect
+
+import kill_handling
+
+class NetworkLossHandler(HandlerBase):
+    alarm_name = 'network_loss'
+
```
Comment (tessbianchi) [inline command/sub8_alarm/sub8_alarm/alarm_handlers/network_loss.py:8]: plans for this?

**Review by tessbianchi — COMMENTED** [review]
Looks generally good, just would like some questions answered.

File: command/sub8_alarm/nodes/event_listeners/keepalive_broadcaster.py (hunk 1 above)
Comment (mattlangford) [reply command/sub8_alarm/nodes/event_listeners/keepalive_broadcaster.py:25]: Correct, it uses header stamps now for timeouts

File: command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py (hunk 2 above)
Comment (mattlangford) [reply command/sub8_alarm/sub8_alarm/alarm_handlers/thruster_out.py:28]: Yep, fixed now.
The param is being passed to the function by the handler so python will complain if there's a missing argument that's not actually there

File: command/sub8_alarm/sub8_alarm/alarm_handlers/network_loss.py (hunk 4 above)
Comment (mattlangford) [reply command/sub8_alarm/sub8_alarm/alarm_handlers/network_loss.py:8]: The `network-loss` alarm triggers a meta kill but maybe something here in the future

**Pull request merged**